### PR TITLE
frontend: add prev & next in build page

### LINKED ISSUE #746: UI: add 'next' and 'previous' build buttons
UI should contain buttons/links to navigate builds without using build list. This feature is meant for single build view only.

### PR DESCRIPTION
**Pull request by amrohassaan**
Fixes #746 adding a previous and next navigation buttons in build page.

### CODE REVIEW

**Comment by amrohassaan** [comment]
Things that are still missing: 
1. Not found builds (in case we reach last build and click next still) returns None and 404 page. 
2. the prev and next url append '?' to url..need to know why and fix.
3. Still missing tests

File: squad/frontend/templatetags/squad.py [hunk 1]
```diff
@@ -91,6 +92,20 @@ def build_url(build):
     return reverse("build", args=(build.project.group.slug, build.project.slug, build.version))
 
 
+@register_global_function
+def previous_build_url(build):
+    prev_bld_status = ProjectStatus.objects.filter(build__project=build.project, build__datetime__lt=build.datetime).last()
```
Comment (amrohassaan) [inline squad/frontend/templatetags/squad.py:97]: I chose to go thru the status here rather than the build just in case more info is needed.

Comment (chaws) [reply squad/frontend/templatetags/squad.py:97]: Any particular reason for `build__project=build.project` instead of `build=build`?

**Comment by chaws** [comment]
It looks good to me, I don't know why "?" keeps appearing in the url though

File: squad/frontend/templatetags/squad.py (hunk 1 above)
Comment (amrohassaan) [reply squad/frontend/templatetags/squad.py:97]: oh wow, forgot am dealing with a db record already :man_facepalming: lol..thanks @chaw :)

**Comment by amrohassaan** [comment]
> for `first()` and `last()` to work properly in this case, I think you will need to specify an explicit ordering in the query (e.g. by datetime)

Makes total sense @terceiro, thanks :)

**Comment by chaws** [comment]
@amrohassaan any updates?

**Comment by amrohassaan** [comment]
> @amrohassaan any updates?

@chaws yes..will finish this today.

File: squad/frontend/templatetags/squad.py [hunk 2]
```diff
@@ -95,6 +95,35 @@ def build_url(build):
...
+        pass
+    if next_build:
+        return build_url(next_build)
+    else:
+        return build_url(build)
+
+
+@register_global_function
+def back_to_latest_build_url(build):
+    return build_url(Build.objects.filter(project=build.project).last())
```
Comment (mwasilew) [inline squad/frontend/templatetags/squad.py:124]: this might get a bit slow when there are a lot of builds. But I think it's OK to take this risk.

**Comment by chaws** [comment]
LGTM

**Pull request merged**